src/banana/toy.py

```diff
@@ -94,8 +53,33 @@ def xfxQ2(self, pid, x, _Q2):
         float
             The value of xf(x,q2).
         """
-
-        return toyLHPDFs(pid, x)
+        xuv = self.N_uv * x ** self.auv * (1e0 - x) ** self.buv
+        xdv = self.N_dv * x ** self.adv * (1e0 - x) ** self.bdv
+        xg = self.N_g * x ** self.ag * (1e0 - x) ** self.bg
+        xdbar = self.N_db * x ** self.adb * (1e0 - x) ** self.bdb
+        xubar = xdbar * (1e0 - x)
+        xs = self.fs * (xdbar + xubar)
+        xsbar = xs
```

Comment:
Instead of these you can directly construct the `self.xpdf` dictionary in the `__init__` method, as a dictionary of callables.
E.g.:

```py
self.xpdf = {}
self.xpdf[21] = lambda x: self.N_g * x ** self.ag * (1e0 - x) ** self.bg
...
self.xpdf[3] = self.xpdf[-3] = lambda x: self.fs * (self.xpdf[-1](x) + self.xpdf[-2](x))
```

and then you simply call it in this method:

```py
return xpdf[pid](x)
```

Reply:
Or you can even define the functions with names as here, that maybe is more readable, and then assign them to the map.